packages/contracts-bedrock/scripts/checks/hotfuzz/main.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+type TestInfo struct {
+	Name     string
+	Contract string
+}
+
+type TestConfig struct {
+	maxSeconds  int
+	maxRuns     int
+	matchPaths  string
+	currentRuns int
+}
+
+func main() {
+	config := parseFlags()
+
+	tests, err := getTestNames(config.matchPaths)
+	if err != nil {
+		log.Fatalf("Error getting test names: %v", err)
+	}
+
+	if len(tests) == 0 {
+		log.Println("No tests found matching the criteria")
+		return
+	}
+
+	var wg sync.WaitGroup
+	results := make(chan string, len(tests))
+
+	for _, test := range tests {
+		wg.Add(1)
+		go runTest(test, config, &wg, results)
+	}
+
+	go func() {
+		wg.Wait()
+		close(results)
+	}()
+
+	for result := range results {
+		fmt.Println(result)
+	}
+}
+
+func parseFlags() TestConfig {
+	maxSeconds := flag.Int("max-seconds", 120, "maximum seconds per test")
+	maxRuns := flag.Int("max-runs", 1000, "maximum number of fuzz runs per test")
+	matchPaths := flag.String("match-path", "", "path pattern to match for test files")
+	flag.Parse()
+
+	return TestConfig{
+		maxSeconds:  *maxSeconds,
+		maxRuns:     *maxRuns,
+		matchPaths:  *matchPaths,
+		currentRuns: 0,
+	}
+}
+
+func getTestNames(matchPaths string) ([]TestInfo, error) {
```

Comment:
Currently this is finding non-fuzz tests too, should filter out anything other than fuzz and invariant tests